tests/functional/api/test_api.py

```diff
@@ -45,3 +45,19 @@ def test_cors_preflight(self, app):
         assert 'POST' in res.headers['Access-Control-Allow-Methods']
         for header in ['Authorization', 'Content-Type', 'X-Client-Id']:
             assert header in res.headers['Access-Control-Allow-Headers']
+
+
+class TestCorsHeaders(object):
+    @pytest.mark.parametrize('url, expect_errors', [
+        # A request that succeeds.
+        ('/api/search', False),
+
+        # A request that triggers a validation error.
+        ('/api/search?sort=raise_an_error', True),
+
+        # A request that fails due to a missing resource.
+        ('/api/annotations/does_not_exist', True),
+    ])
+    def test_responses_have_cors_headers(self, app, url, expect_errors):
```

Comment:
Perhaps amend this test function name so that it indicates that it's testing _error_ responses, specifically?

Reply:
This function is testing _both_ errors and normal responses. Neither had useful integrated tests before.